dictdiffer/__init__.py

```diff
@@ -408,3 +444,79 @@ def revert(diff_result, destination, in_place=False):
                      and return the destination structure.
     """
     return patch(swap(diff_result), destination, in_place)
+
+
+def represent(value):
+    """
+    Return object values such as decimal.Decimal or objects with a __dict__ member in a format that can be
+    reconstructed in patching.
+
+    >>> import decimal
+    >>> represent(decimal.Decimal("1.23"))
+    {'_dictdiffer_value_key': {'type': 'decimal.Decimal', 'value': '1.23'}}
+    >>> import datetime
+    >>> represent(datetime.date(2021, 7, 6))
+    {'_dictdiffer_value_key': {'type': 'datetime.date', 'value': '2021-07-06'}}
+    """
+    transformed_value = False
+    for cls, transform in TRANSFORMS:
```

Comment:
I am not a big fun of global variables. Would it be possible to pass transformations as parameters to `diff` function and propagate them down? I am afraid that the performance will suffer for "dict"-only cases since you will always iterate through this list.